snackpub/Dockerfile

```diff
@@ -0,0 +1,81 @@
+ARG node_version
+
+# Set up dev image
+FROM node:${node_version}-alpine AS dev
+WORKDIR /app
+
+RUN apk add --no-cache --upgrade \
+    bash \
+    busybox \
+    g++ \
+    make \
+    python3
+
+# Set up monorepo
+COPY . ./
+# Install dependencies
+RUN yarn install --frozen-lockfile
+# Build monorepo
+RUN yarn build
+
+# Start snackpub
+WORKDIR /app/snackpub
+CMD ["yarn", "start"]
+
+# Prepare builder for production image without any other sub-packages
+FROM node:${node_version}-alpine AS builder
+WORKDIR /app
+
+RUN apk add --no-cache --upgrade \
+    bash \
+    busybox \
+    g++ \
+    make \
+    python3
```

Comment:
Do we need all of these dependencies like g++ and python3? (Sometimes we get lucky and don't need them if the npm packages we're installing don't need to natively compile Node modules.)

Reply:
these are necessary for the two packages `bufferutil` and `utf-8-validate` to build native code. these two packages are for websocket performance: https://socket.io/docs/v4/performance-tuning/#install-ws-native-add-ons